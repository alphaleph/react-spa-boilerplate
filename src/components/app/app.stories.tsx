import React from 'react';
import { App } from './app';
import '../../index.scss';
import './app.scss';

export const defaultTitle = 'Example Title';

export const defaultStory = () => <App debug={true} title={defaultTitle} />;

export default {
    title: 'App',
    component: App,
    includeStories: ['defaultStory'],
};
