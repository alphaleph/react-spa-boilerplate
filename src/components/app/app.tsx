import React from 'react';
import logo from './logo.svg';
import {
    DarkModeButton,
    PureDarkModeButton,
} from '../dark-mode-button/dark-mode-button';
import './app.scss';

interface AppProps {
    debug?: boolean;
    title: string;
}

const darkModeButtonText = 'Toggle Dark Mode';

export const App = ({ debug = false, title }: AppProps) => {
    return (
        <div className="App">
            <button className="skip-nav test-class">Hello World!</button>
            {debug ? (
                <PureDarkModeButton text={darkModeButtonText} />
            ) : (
                <DarkModeButton text={darkModeButtonText} />
            )}
            <header className="App-header">
                <h1>{title}</h1>
                <img src={logo} className="App-logo" alt="logo" />
                <p>
                    Edit <code>src/app.tsx</code> and save to reload.
                </p>
                <a
                    className="App-link"
                    href="https://reactjs.org"
                    target="_blank"
                    rel="noopener noreferrer"
                >
                    Learn React
                </a>
            </header>
        </div>
    );
};
