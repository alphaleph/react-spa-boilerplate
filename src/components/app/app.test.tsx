import React from 'react';
import { render } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
import { App } from './app';

// const testFn = jest.fn();
const mockTitle = 'Example title';

describe('App', () => {
    // afterEach(() => {
    //   jest.clearAllMocks();
    // });

    it('renders without crashing', () => {
        const { getByText } = render(<App debug={true} title={mockTitle} />);
        const linkElement = getByText(/learn react/i);
        expect(linkElement).toBeInTheDocument();
    });

    it('renders correctly', () => {
        const { getByText } = render(<App debug={true} title={mockTitle} />);
        const linkElement = getByText(/learn react/i);
        expect(linkElement).toHaveAttribute('href', 'https://reactjs.org');
    });

    it('has not changed', () => {
        const { asFragment } = render(<App debug={true} title={mockTitle} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
