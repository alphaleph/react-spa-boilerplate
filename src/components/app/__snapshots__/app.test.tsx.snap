// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has not changed 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <button
      class="skip-nav test-class"
    >
      Hello World!
    </button>
    <button
      class="is-light"
    >
      Toggle Dark Mode
    </button>
    <header
      class="App-header"
    >
      <h1>
        Example title
      </h1>
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <p>
        Edit 
        <code>
          src/app.tsx
        </code>
         and save to reload.
      </p>
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </header>
  </div>
</DocumentFragment>
`;
